File: POPDemo/ViewControllers/FBButtonViewController/FacebookButtonAnimationViewController.m
//
//  FacebookButtonAnimationViewController.m
//  POPDemo
//
//  Created by Hossam Ghareeb on 11/24/14.
//  Copyright (c) 2014 Hossam Ghareeb. All rights reserved.
//

#import "FacebookButtonAnimationViewController.h"

@interface FacebookButtonAnimationViewController ()
@property (weak, nonatomic) IBOutlet UITextField *msgTextField;
@property (weak, nonatomic) IBOutlet UIButton *likeButton;
@property (weak, nonatomic) IBOutlet UIButton *sendButton;

@end

@implementation FacebookButtonAnimationViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}



-(BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string
{
    NSString *comment;
    if(range.length == 0)
    {
        comment = [NSString stringWithFormat:@"%@%@", textField.text, string];
    }
    else
    {
        comment = [textField.text substringToIndex:textField.text.length - range.length];
    }
    
    if (comment.length == 0) {
        //Show like
        [self showLikeButton];
    }
    else
    {
        //Show Send
        [self showSendButton];
    }
    return YES;
}

-(void)showLikeButton
{
    self.likeButton.hidden = NO;
    self.sendButton.hidden = YES;
    
    POPSpringAnimation *spin = [POPSpringAnimation animationWithPropertyNamed:kPOPLayerRotation];
   
    spin.fromValue = @(M_PI / 4);
    spin.toValue = @(0);
    spin.springBounciness = 20;
    spin.velocity = @(10);
    [self.likeButton.layer pop_addAnimation:spin forKey:@"likeAnimation"];
}

-(void)showSendButton
{
    if (self.sendButton.isHidden) {
        
        self.likeButton.hidden = YES;
        self.sendButton.hidden = NO;
        POPSpringAnimation *sprintAnimation = [POPSpringAnimation animationWithPropertyNamed:kPOPViewScaleXY];
        
        sprintAnimation.velocity = [NSValue valueWithCGPoint:CGPointMake(8, 8)];
        sprintAnimation.springBounciness = 20.f;
        [self.sendButton pop_addAnimation:sprintAnimation forKey:@"sendAnimation"];
    }
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Swift Version/POPDemo/Controllers/FBButtonViewController/FacebookButtonAnimationViewController.swift
//
//  FacebookButtonAnimationViewController.swift
//  POPSwiftDemo
//
//  Created by Hossam Ghareeb on 1/29/15.
//  Copyright (c) 2015 AppCoda. All rights reserved.
//

import UIKit

class FacebookButtonAnimationViewController: UIViewController, UITextFieldDelegate {
    
    
    @IBOutlet weak var msgTextField: UITextField!

    @IBOutlet weak var sendButton: UIButton!
    @IBOutlet weak var likeButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
        
        
        var comment : String
        if range.length == 0
        {
            comment = "\(textField.text)\(string)"
        }
        else
        {
            comment = (textField.text as NSString).substringToIndex(countElements(textField.text) - range.length)
            
        }
        
        if countElements(comment) == 0
        {
            showLikeButton()
        }
        else
        {
            showSendButton()
        }
        return true
    }
    
    func showLikeButton()
    {
        
        self.likeButton.hidden = false;
        self.sendButton.hidden = true;
        let spin = POPSpringAnimation(propertyNamed: kPOPLayerRotation)
        spin.fromValue = NSNumber(double: M_PI / 4.0)
        spin.toValue = NSNumber(int: 0)
        spin.springBounciness = 20.0
        spin.velocity = NSNumber(int: 10)
        self.likeButton.layer.pop_addAnimation(spin, forKey: "likeAnimation")
    }
    
    func showSendButton()
    {
        if self.sendButton.hidden
        {
            self.likeButton.hidden = true;
            self.sendButton.hidden = false;
            let spring = POPSpringAnimation(propertyNamed: kPOPViewScaleXY)
            spring.velocity = NSValue(CGPoint: CGPointMake(8, 8))
            spring.springBounciness = 20
            self.sendButton.pop_addAnimation(spring, forKey: "sendAnimation")
        }
    }
    
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
